packages/veritone-react-common/src/components/RangeSlider/index.js

```diff
@@ -0,0 +1,367 @@
+    initialUpperButtonVal: 1
+  };
+
+  static getDerivedStateFromProps(props, state) {
+    const {
+      min,
+      max,
+      dual,
+      value,
+      lowerValue,
+      upperValue,
+      numDecimalDigits,
+      prefix,
+      suffix,
+      showValues
+    } = props;
+
+    // Check for props changed
+    const prevProps = state.prevProps;
+    if (
+      prevProps &&
+      min === prevProps.min &&
+      max === prevProps.max &&
+      value === prevProps.value &&
+      lowerValue === prevProps.lowerValue &&
+      upperValue === prevProps.upperValue &&
+      prefix === prevProps.prefix &&
+      suffix === prevProps.suffix &&
+      showValues === prevProps.showValues &&
+      numDecimalDigits === prevProps.numDecimalDigits
+    ) {
+      return null;
+    }
+
+    const newState = { ...state };
+    newState.prevProps = props;
+
+    // Validate props
+    if (min >= max) {
+      throw new Error('max has to be greater than min');
+    } else if (lowerValue > upperValue) {
+      throw new Error('lower value cannot be greater than upper value');
+    }
+
+    if (dual) {
+      // calculate lower ratio
+      if (lowerValue < min) {
+        newState.lowerRatio = 0;
+        console.warn('lower value is lesser than min');
+      } else if (lowerValue > max) {
+        newState.lowerRatio = 1;
+        console.warn('lower value is greater than max');
+      } else {
+        newState.lowerRatio = (lowerValue - min) / (max - min);
+      }
+
+      // calculate upper ratio
+      if (upperValue < min) {
+        newState.upperRatio = 0;
+        console.warn('upper value is lesser than min');
+      } else if (upperValue > max) {
+        newState.upperRatio = 1;
+        console.warn('upper value is greater than max');
+      } else {
+        newState.upperRatio = (upperValue - min) / (max - min);
+      }
+    } else {
+      newState.lowerRatio = 0;
+      if (value < min) {
+        newState.upperRatio = 0;
+        console.warn('initial value is lesser than min');
+      } else if (value > max) {
+        newState.upperRatio = 1;
+        console.warn('initial value is greater than max');
+      } else {
+        newState.upperRatio = (value - min) / (max - min);
+      }
+    }
+
+    // calculate label width
+    let labelWidth = 0;
+    if (showValues) {
+      labelWidth =
+        Math.round(max).toString().length + prefix.length + suffix.length;
+      numDecimalDigits > 0 && (labelWidth = labelWidth + numDecimalDigits + 1); //
+    }
+
+    // set new initial lower & upper button vals
+    newState.labelWidth = labelWidth + 'ch';
+    newState.lowerValue = dual ? lowerValue : min;
+    newState.upperValue = dual ? upperValue : value;
+    newState.initialLowerButtonVal = newState.lowerRatio;
+    newState.initialUpperButtonVal = newState.upperRatio;
+
+    return newState;
+  }
+
+  handleMouseDown = event => {
+    document.onmousemove = this.handleMouseMove;
+    document.onmouseup = this.handleMouseUp;
```

Comment:
sketchy -- what if two components in an app wanted to both use document listeners like this? They'd be constantly clobbering each other. 

You should only be dealing with listeners within React itself (as props on an `<Element>`). This would probably be better done using html draggable/ondrag than global listeners?